Sources/Amplitude/Plugins/AmplitudeDestinationPlugin.swift

```diff
@@ -7,11 +7,12 @@
 
 public class AmplitudeDestinationPlugin: DestinationPlugin {
     private var pipeline: EventPipeline?
+    private var identifyInterceptor: IdentifyInterceptor?
 
     internal func enqueue(event: BaseEvent?) {
-        if let e = event {
-            if e.isValid() {
-                pipeline?.put(event: e)
+        if let event {
+            if event.isValid() {
+                identifyInterceptor?.intercept(event: event)
```

Comment:
I still prefer the implementation in the other SDKs where `identifyInterceptor?.intercept()` will consume the event if it is intercepted, or apply to intercepted values to the current event, since it allows the existing pipeline logic to work the same as before and limit the scope of  IdentifyInterceptor to only handle 1) intercept + store intercept and 2) decorate the given event with intercept data.

Reply:
1. In this case `pipeline.put(event)` logic is spread between the two classes. I think, ideally `pipeline.put(event)` logic should be in one of the classes.
2. If we use pattern like
```
mergedEvent = mergeEvent(event, getCollectedIdentifyEvents) // and clear collected identify events in getCollectedIdentifyEvents
pipeline.put(mergedEvent) // vs return event
```
there is a chance the all collected identify data can be lost if app/device crashes between 1 and 2 steps
I think, something like next snippet looks more stable:
```
mergedEvent = mergeEvent(event, getCollectedIdentifyEvents)
pipeline.put(mergedEvent)
removeCollectedIdentifyEvents
```

Reply:
Thanks, this is a good call out. We should probably do something like #2. I can work on this today.